src/plugins/data_source_management/public/plugin.ts

```diff
@@ -303,10 +319,85 @@ export class DataSourceManagementPlugin
     };
     setRenderAssociatedObjectsDetailsFlyout(renderAssociatedObjectsDetailsFlyout);
 
+    const getDirectQuerySyncFromUrl = (): boolean | null => {
+      const parsedHash = parseUrlHash(window.location.href);
+      if (parsedHash && parsedHash.query) {
+        const directQuerySync = parsedHash.query.directQuerySync;
+        if (directQuerySync === 'true') return true;
+        if (directQuerySync === 'false') return false;
+      }
+      return null;
+    };
+
+    const shouldMountBanner = (): boolean => {
+      const urlFlag = getDirectQuerySyncFromUrl();
+      if (urlFlag !== null) {
+        return urlFlag;
+      }
+      return this.config.dashboardDirectQuerySyncEnabled;
+    };
+
+    const updateBanner = () => {
+      const appId = this.currentAppId;
+
+      if (appId === 'dashboards') {
+        const hash = window.location.hash;
+        const isDashboardViewMatch = hash.match(/#\/view\/([^\/?]+)(\?.*)?$/);
+        if (isDashboardViewMatch && isDashboardViewMatch[1]) {
+          const dashboardId = isDashboardViewMatch[1];
+          if (shouldMountBanner()) {
+            if (!this.bannerId) {
+              this.bannerId = core.overlays.banners.add(
+                toMountPoint(
+                  React.createElement(DashboardDirectQuerySync, {
+                    dashboardId,
+                    http: core.http,
+                    notifications: core.notifications,
+                    savedObjectsClient: core.savedObjects.client,
+                    removeBanner: () => {
+                      if (this.bannerId) {
+                        core.overlays.banners.remove(this.bannerId);
+                        this.bannerId = null;
+                      }
+                    },
+                  })
+                )
+              );
+            }
+          } else if (this.bannerId) {
+            core.overlays.banners.remove(this.bannerId);
+            this.bannerId = null;
+          }
+        } else if (!isDashboardViewMatch && this.bannerId) {
+          core.overlays.banners.remove(this.bannerId);
+          this.bannerId = null;
+        }
+      } else if (this.bannerId) {
+        core.overlays.banners.remove(this.bannerId);
+        this.bannerId = null;
+      }
+    };
+
+    core.application.currentAppId$.subscribe((appId: string | undefined) => {
+      this.currentAppId = appId;
+      updateBanner();
+    });
+
+    window.addEventListener('hashchange', () => {
+      updateBanner();
+    });
+
+    updateBanner();
+
     return {
       getAuthenticationMethodRegistry: () => this.authMethodsRegistry,
     };
   }
 
-  public stop() {}
+  public stop() {
+    if (this.bannerId && this.core) {
+      this.core.overlays.banners.remove(this.bannerId);
```

Comment:
is coreStart still available when stop is triggered?

Reply:
Yes, `this.core` (a `CoreStart` instance) is available when `stop` is triggered. I think in the lifecycle, `start` is called before `stop`, and `start` sets `this.core` to the provided `CoreStart` instance. Since `stop` is only invoked on a started plugin, `this.core` will always be non-`null` at that point. 

Additionally, the `if (this.bannerId && this.core)` check provides extra safety, though it’s not strictly necessary given the lifecycle guarantee.